io.openems.edge.ess.generic/src/io/openems/edge/ess/generic/symmetric/GenericManagedSymmetricEssImpl.java

```diff
@@ -212,9 +220,17 @@ public int getPowerPrecision() {
 		if (error != null) {
 			throw error; // if any error happened, throw the last one
 		}
-		return result;
-	}
 
+		// If the GenericEss is not in State "STARTED" block ACTIVE and REACTIVE Power!
+		if (!this.isStarted()) {
+			result.add(this.createPowerConstraint("ActivePower Contraint ESS not Started", Phase.ALL, Pwr.ACTIVE,
+					Relationship.EQUALS, 0));
+			result.add(this.createPowerConstraint("ReactivePower Contraint ESS not Started", Phase.ALL, Pwr.REACTIVE,
+					Relationship.EQUALS, 0));
+		}
+		return result.toArray(new Constraint[] {});
```

Comment:
Diese Anweisung funktioniert nur zufällig; das musste ich jetzt selbst in den Javadocs nachlesen: https://docs.oracle.com/javase/8/docs/api/java/util/List.html#toArray-T:A-

Ich schlage entweder
`return result.toArray();`
oder
`return result.toArray(new Constraint[result.size()]);`
vor, um die unnötige Erzeugung eines leeren Arrays zu vermeiden.
(Beispiel: https://devqa.io/convert-list-to-array-in-java/)